contract/src/swaparoo.contract.js

```diff
@@ -121,6 +121,17 @@ export const start = async (zcf, privateArgs, baggage) => {
   // TODO: update with Fee param
   const feeShape = makeNatAmountShape(feeBrand, params.getFee().value);
 
+  const generateOfferNonce = (() => {
+    // XXX: This is not stored durably, so ensure it stays unique after upgrade,
+    // perhaps by appending a new character each time.
+    let offerNonce = -1;
+
+    return () => {
+      offerNonce += 1;
+      return `${offerNonce}`;
+    };
+  })();
```

Comment:
Since this is an example, we might as well store it durably; adding it isn't any longer than the XXX comment:

```suggestion
  const generateOfferNonce = (() => {
    let offerNonce = provide(baggage, 'offerNonce', () => -1);

    return () => {
      offerNonce += 1;
      baggage.set('offerNonce', offerNonce);
      return `${offerNonce}`;
    };
  })();
```

along with:

```
import { provide } from '@agoric/vat-data';
```

Reply:
Also, we might as well `return offerNonce`, since it gets stringified later.

Reply:
Done. Just writing out my discovery process for reference (tried not going straight to your hints):
- Looked for other durable examples in `dapp-agoric-basics`, found `makeDurableGovernorFacet`, doesn't seem like quite what I want.
- Go to docs.agoric.com and search "durability", find https://docs.agoric.com/guides/zoe/contract-upgrade.html#durability
- See an example with `provide` and `makeScalarBigMapStore`. I just need a number, not a map, not sure if that's allowed, but from your example it apparently is.
- `yarn add @agoric/vat-data` and use the `provide` method -> Tests failing, seems like it brought in some conflicting dependency versions.
- Nuke `node_modules`, remove `@agoric/vat-data`, do another `yarn install` and then do `yarn why @agoric/vat-data` to see which version is already transitively depended on.
- Add that version to `package.json`, now only one test case failing:
```
build-proposal › bundles from build:deployer meet 1MB request limit
  test/test-build-proposal.js:18

   17:     const rootPath = undefined; // not needed since bundle is already cached
   18:     const bundle = await bundleCache.load(rootPath, name);
   19:     const fileContents = JSON.stringify(bundle);          

  Rejected promise returned by test. Reason:

  TypeError {
    code: 'ERR_INVALID_ARG_TYPE',
    message: 'The "paths[1]" argument must be of type string. Received undefined',
  }
```
- Some hint about bundle caching there, run `yarn build` before `yarn test` again and now it passes.

Reply:
> ... writing out my discovery process...

nice!

> * `yarn add @agoric/vat-data` and use the `provide` method -> Tests failing, seems like it brought in some conflicting dependency versions.
> * Nuke `node_modules`, remove `@agoric/vat-data`, do another `yarn install` and then do `yarn why @agoric/vat-data` to see which version is already transitively depended on.

Yes, adding dependencies is incredibly fragile. I'm tracking that as...

 - https://github.com/Agoric/agoric-sdk/issues/8621